train.py

```diff
@@ -0,0 +1,99 @@
+import os
+import sys
+import time
+import argparse
+
+import torch
+import psutil
+import imageio
+import numpy as np
+from torchvision.utils import save_image
+
+from utils import get_interpolations
+from models.autoencoder import Autoencoder
+from train_utils import test_epoch, train_epoch
+
+parser = argparse.ArgumentParser(description="Main function to call training for different AutoEncoders")
+parser.add_argument("--batch-size", type=int, default=128, metavar="N", help="input batch size for training (default: 128)")
+parser.add_argument("--epochs", type=int, default=10, metavar="N", help="number of epochs to train (default: 10)")
+parser.add_argument("--no-cuda", action="store_true", default=False, help="enables CUDA training")
+parser.add_argument("--seed", type=int, default=42, metavar="S", help="random seed (default: 1)")
+parser.add_argument(
+    "--log-interval", type=int, default=10, metavar="N", help="how many batches to wait before logging training status"
+)
+parser.add_argument(
+    "--embedding-size", type=int, default=32, metavar="N", help="how many batches to wait before logging training status"
+)
+parser.add_argument("--results_path", type=str, default="results/", metavar="N", help="Where to store images")
+parser.add_argument("--model", type=str, default="AE", metavar="N", help="Which architecture to use")
+parser.add_argument("--dataset", type=str, default="MNIST", metavar="N", help="Which dataset to use")
+
+args = parser.parse_args()
+args.cuda = not args.no_cuda and torch.cuda.is_available()
+torch.manual_seed(args.seed)
+
+ae = Autoencoder(args)
+architectures = {"AE": ae}
+
+print(args.model)
+
+if __name__ == "__main__":
+    try:
+        os.stat(args.results_path)
+    except FileNotFoundError:
+        os.mkdir(args.results_path)
```

Comment:
```suggestion
    os.mkdirs(args.results_path, exist_ok=True)
```

In general, I think it's better to prefer if statements over try-except blocks. In this `mkdirs` should be enough